docs/key_modules/memory.md

```diff
@@ -0,0 +1,315 @@
+- `storage`: Optional BaseKeyValueStorage
+- `window_size`: Optional int for retrieval window
+
+Methods:
+
+- `retrieve()`: Get recent chat messages
+- `write_records()`: Write new records to chat history
+- `get_context_creator()`: Get the context creator
+- `clear()`: Remove all chat messages
+
+### VectorDBMemory
+
+An AgentMemory implementation that wraps VectorDBBlock.
+
+Initialization:
+
+- `context_creator`: BaseContextCreator
+- `storage`: Optional BaseVectorStorage
+- `retrieve_limit`: int for maximum number of retrieved messages (default: 3)
+
+Methods:
+
+- `retrieve()`: Get relevant messages from the vector database
+- `write_records()`: Write new records and update current topic
+- `get_context_creator()`: Get the context creator
+
+### LongtermAgentMemory
+
+Combines ChatHistoryMemory and VectorDBMemory for comprehensive memory management.
+
+Initialization:
+
+- `context_creator`: BaseContextCreator
+- `chat_history_block`: Optional ChatHistoryBlock
+- `vector_db_block`: Optional VectorDBBlock
+- `retrieve_limit`: int for maximum number of retrieved messages (default: 3)
+
+Methods:
+
+- `retrieve()`: Get context from both chat history and vector database
+- `write_records()`: Write new records to both chat history and vector database
+- `get_context_creator()`: Get the context creator
+- `clear()`: Remove all records from both memory blocks
+
+## Usage
+
+To use the Memory module in your agent:
+
+1. Choose an appropriate AgentMemory implementation (ChatHistoryMemory, VectorDBMemory, or LongtermAgentMemory).
+2. Initialize the memory with a context creator and any necessary parameters.
+3. Use `write_records()` to add new information to the memory.
+4. Use `retrieve()` to get relevant context for the agent's next action.
+5. Use `get_context()` to obtain the formatted context for the agent.
+
+Example using LongtermAgentMemory:
+
+```python
+from camel.memories import LongtermAgentMemory, ChatHistoryBlock, VectorDBBlock, ScoreBasedContextCreator
+from camel.memories import MemoryRecord
+from camel.messages import BaseMessage
+from camel.types import OpenAIBackendRole
+from camel.utils import OpenAITokenCounter
+
+# Initialize the memory
+memory = LongtermAgentMemory(
+    context_creator=ScoreBasedContextCreator(OpenAITokenCounter, 1024),
+    chat_history_block=ChatHistoryBlock(),
+    vector_db_block=VectorDBBlock()
+)
+
+# Create and write new records
+records = [
+    MemoryRecord(
+        message=BaseMessage(role_name="user", role_type="human", meta_dict=None, content="What is machine learning?"),
+        role_at_backend=OpenAIBackendRole.USER
+    ),
+    MemoryRecord(
+        message=BaseMessage(role_name="agent", role_type="ai", meta_dict=None, content="Machine learning is a subset of AI..."),
+        role_at_backend=OpenAIBackendRole.ASSISTANT
+    )
+]
+memory.write_records(records)
+
+# Get context for the agent
+context, token_count = memory.get_context()
+
+# Use the context in your agent's next action
```

Comment:
This code block looks the same as the previous one to me. It may be more interesting to showcase how to use `LongtermAgentMemory` with `ChatAgent`.